File: Practicas Teoricas/practicas/bytecode_ej_e/ack.c
#include <stdio.h>

// Función de Ackermann
int ackermann(int m, int n) {
    if (m == 0) {
        return n + 1;
    } else if (n == 0) {
        return ackermann(m - 1, 1);
    } else {
        return ackermann(m - 1, ackermann(m, n - 1));
    }
}

int main() {
    printf("ackermann en 3 11 es %d\n", ackermann(3, 11));
    return 0;
}

File: Practicas Teoricas/practicas/bytecode_ej_e/ack.hs
ackermann :: Int -> Int -> Int
ackermann m n = 
    if m == 0 
        then n + 1 
        else 
            if n == 0 
                then ackermann (m - 1) 1 
                else ackermann (m - 1) (ackermann m (n - 1))
